src/sage/matroids/matroid.pyx

```diff
@@ -3177,8 +3385,12 @@ cdef class Matroid(SageObject):
         ambient = FreeModule(ZZ, n)
         vector_e = ambient.basis()
         convert = {ind: i for i, ind in enumerate(self.groundset())}
-        vertices = [ambient.sum(vector_e[convert[i]] for i in IS)
-                    for IS in self.independent_sets()]
+        vertices = []
+        for IS in self.independent_sets_iterator():
+            lst = []
+            for i in IS:
+                lst += [vector_e[convert[i]]]
+            vertices += [ambient.sum(lst)]
```

Comment:
```suggestion
                lst.append(vector_e[convert[i]])
            vertices.append(ambient.sum(lst))
```
Not sure if this is faster, but it could be. IMO, it is easier to understand the intent of adding a single element to the lists.